jni/src/knn_extension/faiss/utils/BitSet.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include "BitSet.h"
+
+FixedBitSet::FixedBitSet(const int* int_array, const int length){
+    assert(int_array && "int_array should not be null");
+    int maxValue = int_array[length - 1];
+    this->numBits = (maxValue >> 6) + 1; // div by 64
+    this->bitmap = new uint64_t[this->numBits]();
+    for(int i = 0 ; i < length ; i ++) {
+        int value = int_array[i];
+        int bitsetArrayIndex = value >> 6;
+        this->bitmap[bitsetArrayIndex] |= 1ULL << value; // Equivalent of 1ULL << (value % 64)
```

Comment:
over here the assumption is there are no duplicates in int_array. Should we add this in documentation too.

Reply:
BitSet is set which does not allow duplication in general. I think it is too obvious to document it.